include/highfive/h5easy_bits/H5Easy_vector.hpp

```diff
@@ -17,31 +17,30 @@ namespace H5Easy {
 
 namespace detail {
 
-namespace vector {
+template<class T> struct is_vector: std::false_type  {};
+template<class T> struct is_vector<std::vector<T>> : std::true_type {};
 
 using HighFive::details::type_of_array;
 using HighFive::details::get_dim_vector;
 
-// create DataSet and write data
-template <class T, class E = void>
-struct dump_impl
-{
-    static DataSet run(File& file, const std::string& path, const std::vector<T>& data)
-    {
-        using type_name = typename type_of_array<T>::type;
-        detail::createGroupsToDataSet(file, path);
-        DataSet dataset = file.createDataSet<type_name>(path, DataSpace::From(data));
-        dataset.write(data);
-        file.flush();
-        return dataset;
-    }
-};
-
-// replace data of an existing DataSet of the correct size
-template <class T, class E = void>
-struct overwrite_impl
-{
-    static DataSet run(File& file, const std::string& path, const std::vector<T>& data)
+/*
+This vector specialization does not implement load, load_part, dump_extend.
+Since it inherits from io_impl_base, implementations there will be called automatically.
+ */
+template<typename T>
+struct io_impl<T,typename std::enable_if<is_vector<T>::value>::type>: public io_impl_base<T> {
```

Comment:
If I understand well, a nested vector is also capture here, right?

Reply:
Yes, of course. The idea is that e.g. `vector<set<T>>` will fail in the scalar routine called with `set<T>`.